src/test/java/net/kaczmarzyk/spring/data/jpa/domain/BetweenTest.java

```diff
@@ -188,4 +190,12 @@ public void rejectsTooFewArguments() throws ParseException {
     public void rejectsTooManyArguments() throws ParseException {
         new Between<>(queryCtx, "path", new String[] { "2014-03-10", "2014-03-11", "2014-03-11" }, defaultConverter);
     }
+
+	@Test
+	public void equalsContract() {
```

Comment:
general comment: I think it's a good idea to unit test equal behaviour of cglib-enhanced classes.